src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateStep.java

```diff
@@ -48,6 +51,26 @@ public void setUpstream(String upstream) {
         this.upstream = fixEmptyAndTrim(upstream);
     }
 
+    @CheckForNull
+    public BuildSelector getSelector() {
+        return selector;
+    }
+
+    @DataBoundSetter
+    public void setSelector(BuildSelector selector) {
+        this.selector = selector;
+    }
+
+    @CheckForNull
+    public String getParameters() {
+        return parameters;
+    }
+
+    @DataBoundSetter
+    public void setParameters(String parameters) {
```

Comment:
Indeed, the parameters are something that should be bound to the selector.
The current *BuildSelector* API may not be very good, as you said. The API will be changed if moved to a separate repository.